src/tempstorage/TemporaryStorage.py

```diff
@@ -324,6 +324,7 @@ def _takeOutGarbage(self, oid):
         for k in self._conflict_cache.keys():
             if k[0] == oid:
                 del self._conflict_cache[k]
+                break
```

Comment:
I'm not convinced the fix is right.  The data structure is a plain `dict` mapping (oid, serial) to (data, timestamp).

The old code would remove all cache keys that had oid as the first element.  The new code removes only the first key.

The old code worked because on Python 2 you can safely modify the dictionary while iterating over its `.keys()`.  The straightforward fix for Python 3 would be to change the loop to loop over `list(self._conflict_cache.keys())`.

Reply:
You're right. As we are searching for a value in the dict. This won't do.